Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager_test.py

```diff
@@ -0,0 +1,999 @@
+        ({"a": 1}, {"b": 2}, {"a": "A"}, {"A": 1, "b": 2}),
+    ],
+    ids=[
+        "add args to empty  kwargs",
+        "keep kwargs the same, when no args",
+        "add args to kwargs that already have values",
+    ],
+)
+def test_update_if_value(
+    args: dict[str, str],
+    kwargs: dict[str, str],
+    input_to_output_keys: dict[str, str],
+    expected_results: dict[str, str],
+):
+    assert update_if_value(args, kwargs, input_to_output_keys) == expected_results
+
+
+@pytest.mark.parametrize(
+    ("value", "excepted_value"),
+    [
+        ("key1:value1", {"key1": ["value1"]}),
+        ("key1:value1,", {"key1": ["value1"]}),
+        ("key1:value1,value2", {"key1": ["value1", "value2"]}),
+        ("key1:value1,value2,", {"key1": ["value1", "value2"]}),
+        (
+            "key1:value1,value2,key2:value1",
+            {"key1": ["value1", "value2"], "key2": ["value1"]},
+        ),
+        (
+            "key1:value1,value2,key2:value1,",
+            {"key1": ["value1", "value2"], "key2": ["value1"]},
+        ),
+        (
+            "key1:value1,key2:value1,value2,value3,key3:value1",
+            {
+                "key1": ["value1"],
+                "key2": ["value1", "value2", "value3"],
+                "key3": ["value1"],
+            },
+        ),
+        (
+            "key1:value1,key2:value1,value2,value3,key3:value1,",
+            {
+                "key1": ["value1"],
+                "key2": ["value1", "value2", "value3"],
+                "key3": ["value1"],
+            },
+        ),
+    ],
+)
+def test_format_parameters_arguments(
+    value: str,
+    excepted_value: dict[str, Any],
+) -> None:
+    """Given
+        a dictionary of parameters,
+    When
+        `format_parameters_arguments` is called with the parameters,
+    Then
+        it should return the expected formatted response.
+    """
+    assert format_parameters_arguments(value) == excepted_value
+
+
+@pytest.mark.parametrize(
+    ("document_version", "expected_response"),
+    [
+        ("test_version", "test_version"),
+        ("default", "$DEFAULT"),
+        ("latest", "$LATEST"),
+    ],
+    ids=["custom version", "default version", "latest version"],
+)
+def test_format_document_version(document_version: str, expected_response: str) -> None:
+    """Given
+        a document version,
+    When
+        `format_document_version` is called with the document version,
+    Then
+        it should return the expected formatted response.
+    """
+    assert format_document_version(document_version) == expected_response
+
+
+@pytest.mark.parametrize(
+    ("args", "expected_error_message"),
+    [
+        ({"instance_id": "test_id"}, "Invalid instance id: test_id"),
+        ({"association_id": "test_id"}, "Invalid association id: test_id"),
+        (
+            {"association_version": "test_version"},
+            "Invalid association version: test_version",
+        ),
+    ],
+    ids=[
```

Comment:
imo, easier to read when `id=` is provided after each `pytest.param`, than as a separate list